be/src/storage/rowset/cast_column_iterator.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "column/chunk.h"
+#include "column/vectorized_fwd.h"
+#include "common/macros.h"
+#include "storage/rowset/column_iterator_decorator.h"
+
+namespace starrocks {
+
+class Column;
+class ColumnRef;
+class Expr;
+class ObjectPool;
+
+class CastColumnIterator : public ColumnIteratorDecorator {
+public:
+    // Will take the ownership of |source_iter|
```

Comment:
is it possible to require the `source_iter` to be a unique_ptr, explicitly expressing the acquire of the ownership.

Reply:
updated